### DISCUSSION TITLE
How to add Player Image?

### DISCUSSION THREAD

**Discussion by luckygumber** [scribble-rs/scribble.rs#282] [Q&A]
I am thinking about extending this project but I want to include a character (display) image. The user can select one of the available images when creating the lobby and can also edit it after joining a game (The edit username dialog would be modified for this). The frontend changes I can probably do but am confused about the API / backend changes. Can anyone guide me on this?

**Comment by Bios-Marcel** [comment]
If we are talking about a fixed set of images that all have a unique name, i'd just add a public field to the `Player` struct with a `string` describing the image name. The client can then access the file from the servers `/resources` endpoint or however else it pleases, in case it's a third party client. Either way, I'd say you shouldn't add the image choice at lobby creation and just do it in the lobby. The additional complication just for the lobby creator isn't worth it and makes future changes and refactors more difficult.

**Reply by luckygumber** [reply]
Yes, there would be a fixed set of images all with a unique name in the `/resources` endpoint. I have added a `DisplayImage` (string) field to the `Player` struct. I am trying some changes. Will update here after it's done.

**Reply by luckygumber** [reply]
Changes required to implement this:

- In **Frontend/create.go**: Add a new variable to get the image name from create lobby home page form and pass it to to the _game.createLobby()_ function
- In **Game/data.go**: Add new `string` in Player struct to store the player's image as `string` and pass new argument to _createPlayer()_ function which will set this value
- In **Game/lobby.go**: Pass new argument `playerImage` to _createLobby()_ function, which will then pass it to _createPlayer()_ function
